static/js/pages/certificates/CertificateGenerate.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { FC, useState } from "react";
+import { generateCert } from "util/certificate";
+import { Button, Col, Icon, Row } from "@canonical/react-components";
+import BaseLayout from "components/BaseLayout";
+import BrowserImport from "pages/certificates/BrowserImport";
+
+interface Certs {
+  crt: string;
+  pfx: string;
+}
+
+const CertificateGenerate: FC = () => {
+  const [certs, setCerts] = useState<Certs | null>(null);
+
+  const createCert = () => {
+    const certs = generateCert();
```

Comment:
Currently the UI gets frozen for a few seconds. I know that there is a warning to the user that this might happen, but usually it is better to guard this scenario with a loader.